gdsfactory/pdk.py

```diff
@@ -709,6 +709,19 @@ def get_layer_tuple(layer: LayerSpec) -> tuple[int, int]:
     return info.layer, info.datatype
 
 
+def get_layer_info(layer: LayerSpec) -> kf.kdb.LayerInfo:
+    """Returns layer info from a layer spec."""
+    layer_index = get_layer(layer)
+    if isinstance(layer_index, LayerEnum):
+        return kf.kcl.get_info(layer_index.value)  # type: ignore[no-any-return]
+    elif isinstance(layer_index, int):
+        return kf.kcl.get_info(layer_index)  # type: ignore[no-any-return]
```

Comment:
You don't need to check separately `LayerEnum` will pass the check `isinstance(enumvariable, int)`